### ISSUE TITLE
Received pictures auto picked up and saved by Google Photos

### ISSUE THREAD

**Issue by JoshuaGarrison27** [klinker-apps/messenger-issues#408]
Device with issue: Huawei, Nexus 6P, Stock Version 7.1.2
Primary device: Huawei, Nexus 6P, Stock Version 7.1.2
App version: 2.4.9.1391

**Issue**
Im not sure that this is a Pulse issue or a Google Photos issue. But I have checked to see if I have the directory set up to auto upload photos and I do not. I cannot figure out how to make it stop auto uploading all gifs and photos that I receive.  I don't want to turn off all auto uploads in Google Photos. I just want to stop uploading thing I receive in Pulse unless I explicitly download them.

**Steps to reproduce:**
Receive an MMS image from anyone

**Additional information:**
No additional information at this time. I will provide anything needed and within reason upon request.

**Comment by klinker24** [comment]
Do you have the auto save media option on? I don't do anything special, but when an MMS is saved to your storage, it will be picked up by gallery apps.

You should turn off the auto save media option if you don't want this to happen. Messages will still be downloaded to Pulse, just not saved to your storage.

**Comment by JoshuaGarrison27** [comment]
I have the auto-save turned off. You mean in the Global Settings > MMS Configuration > Auto-save media, right? I have that turned off. The Media Download Directory is set to a folder that I know for a fact does not get uploaded to Google Photos.

**Comment by klinker24** [comment]
So they only get uploaded when you go into Pulse and use the save button there? That definitely seems like a Google photos issue to me, yes..

**Comment by JoshuaGarrison27** [comment]
No it happens when I just receive the MMS. I dont manually download the image or have auto-save turned on in the settings. It just happens on its own. I have check the Google Photos settings and Pulse settings and I cannot figure out why it is still happening.

**Comment by klinker24** [comment]
Hmm, I have a feeling that you have another SMS app that is auto saving messages to storage then, and Google Photos is picking them up. I don't think that could possibly be coming from Pulse... Hmm

**Comment by JoshuaGarrison27** [comment]
Do you think it could be the stock SMS app? Besides the stock 'Messages' app, Pulse is the the only actual SMS application that I have. However, with that said, if I go to change the default, I have the default 'Messages', 'Pulse', 'Messenger' (the Facebook app, but I have the SMS feature turned off), 'SMS Backup & Restore'.

You very well could be right. It could be the backup program or Facebook not respecting my settings to not interfere with SMS.

My other thought is, could it be Android automatically storing the images?

Would you like for me to uninstall Facebook Messenger and SMS Backup/Restore to see if that solves the issue or try some other form of debugging?

**Comment by klinker24** [comment]
Google photos will display the location of the local photo. I say that you change the "Media Download Directory" in Pulse's settings, and see what directory the images are coming from. It could very well be the backup and restore app if that app is backing up messages regularly or automatically.

If you aren't saving those images through Pulse, then I am pretty confident that my app is not where they are getting uploaded from.

**Comment by JoshuaGarrison27** [comment]
So I am looking at the location for each photo that is in my library and looking at the storage location. For the ones that I took it shows a location on the device. However, for images sent to me it just just shows something like "mmsattachment=:=12655" (where the 12655 is a varying number).

**Comment by klinker24** [comment]
I would try without the other apps installed. Like I said, pretty confident that has nothing to do with Pulse.

**Comment by JoshuaGarrison27** [comment]
Okay I will give it a shot. I will report back here with the results.

**Comment by JoshuaGarrison27** [comment]
Just an update: I uninstalled the facebook app and SMS backup. I even disabled the stock messenger app.Had a buddy of mine send me a photo and it still was uploaded to Google Drive. So now I am beginning to wonder if it has something to do with [Join](https://play.google.com/store/apps/details?id=com.joaomgcd.join&hl=en). The problem is if I uninstall Join I will lose my device api key and break some of my automation. I am going to reach out to the developer and see if he might think it is his application. I will report back here soon.

**Comment by JoshuaGarrison27** [comment]
Okay. Found the root problem. It is in fact a [Join](https://play.google.com/store/apps/details?id=com.joaomgcd.join&hl=en) issue.

Here is the details:
So the issue is that a compressed version of the image file is being added to Google Drive for the push notification to all devices. These images are then getting picked up and showing in Google Photos. I dont know if there is a way to exclude a folder from showing in Google Photos...

[Google+ Link to the Join discussion](https://plus.google.com/u/0/112239156454675366879/posts/WoqSRLLKQCL)

Sorry for the confusion and any wasted time or effort on your part. Thank you very much for your support @klinker24 in diagnosing this issue. Maybe the notes we have compiled here will help someone down the road.
